File: src/main/java/com/example/demoMyBatis/service/UserService.java
package com.example.demoMyBatis.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.demoMyBatis.mapper.UserMapper;
import com.example.demoMyBatis.model.User;

@Service
public class UserService{

    @Autowired
    private UserMapper userMapper;

    public List<User> getAllUsers(){
        return userMapper.getAllUsers();
    }

    public User getUserById(int id){
        return userMapper.getUserById(id);
    }

    public void createUser(User user){
        userMapper.insertUser(user);
    }

    public void updateUser(User user){
        userMapper.updateUser(user);
    }

    public void deleteUser(int id){
        userMapper.deleteUser(id);
    }

    public List<User> getAllUserByNameContains(String name){
        return userMapper.getAllUserByNameContains(name);
    }

}

File: src/service/UserService.js
import axios from "axios";

const API_URL = 'http://localhost:8080/users';

class UserService{

    getAllUsers(){
        return axios.get(API_URL);
    }

    getUserById(id){
        return axios.get(`${API_URL}/${id}`);
    }

    createUser(users){
        return axios.post(API_URL, users);
    }

    updateUser(id, user){
        return axios.put(`${API_URL}/${id}`, user);
    }

    deleteUser(id){
        return axios.delete(`${API_URL}/${id}`);
    }

    getAllUserByNameContains(name){
        return axios.get(`${API_URL}/findByName`, { params: {name} });
    }
}

export default new UserService();